src/hooks/useDSHeading.tsx

```diff
@@ -0,0 +1,37 @@
+import Heading from "../components/Heading/Heading";
+
+/**
+ * DS internal helper hook to render a default `h2` heading element if the
+ * passed title is a string. Otherwise, it will return the title as is if it
+ * is a JSX element.
+ */
+function useDSHeading({
+  title = "",
+  id = "",
+  customDefaultHeading,
+}: {
+  title: string | JSX.Element;
+  id: string;
```

Comment:
The `id` prop for the `Heading` component is optional.  Why do you make it required?

Reply:
Yes, it should be optional and didn't get to that just yet. I noticed for some components, the optional `id` value is used in other ids. This makes it render as `undefined`. In the `List` component, for example, the internal heading has an id of `${id}-heading` but since `id` is optional, it will render as `undefined-heading`. That's something I wanted to tackle in an update in this PR after feedback.

Reply:
Right.  That is probably leftover from when we did have all the `id` props set as required.